src/felix86/v2/handlers.cpp

```diff
@@ -5741,6 +5741,145 @@ FAST_HANDLE(BSR) {
     as.Bind(&end);
 }
 
+void LZCNT(Recompiler& rec, Assembler& as, biscuit::GPR result, biscuit::GPR src, int size) {
+    if (Extensions::B) {
+        // hardware CLZ
+        if (size == 64) {
+            as.CLZ   (result, src);
+        } 
+        else if (size == 32) {
+            as.CLZW  (result, src);
+        } 
+        else if (size == 16) {
+            as.CLZW(result, src);
+            as.ADDI(result, result, -16);
+        } 
+        else if (size == 8) {
+            as.CLZW  (result, src);
+            as.ADDI(result, result, -24);
+        } 
+        else {
+            UNREACHABLE();
+        }
+    } else {
+        // software fallback: scan down from MSB until we see a 1
+        // we’ll keep a 'count' in result, and a temporary in 'scratch'
+        biscuit::GPR scratch = rec.scratch();
+        as.LI   (result, 0);                     // count = 0
+        Label loop, done;
+        as.Bind (&loop);
+          // compute shift = (size-1) - count
+          as.LI   (scratch, size - 1);
+          as.SUB  (scratch, scratch, result);
+          // test bit: (src >> shift) & 1
+          as.SRL  (scratch, src, scratch);
+          as.ANDI (scratch, scratch, 1);
+          as.BNEZ (scratch, &done);
+          // still zero, bump count and retry
+          as.ADDI (result, result, 1);
+          as.J    (&loop);
+        as.Bind (&done);
+        rec.popScratch();
+    }
+}
+
+FAST_HANDLE(LZCNT) {
+    biscuit::GPR src    = rec.getGPR(&operands[1]);
+    biscuit::GPR result = rec.scratch();
+
+    // perform the count (stub above handles both hw and sw paths)
+    LZCNT(rec, as, result, src, instruction.operand_width);
+
+    // write back the destination register
+    rec.setGPR(&operands[0], result);
+
+    // ZF ← (result == size)  <=>  src was zero
+    if (rec.shouldEmitFlag(rip, X86_REF_ZF)) {
+      biscuit::GPR zf     = rec.flag(X86_REF_ZF);
+      as.SEQZ  (zf,  result);            // ZF = (tmp == 0)
+    }
+
+    if (rec.shouldEmitFlag(rip, X86_REF_CF)) {
+      biscuit::GPR cf     = rec.flag(X86_REF_CF);
+      biscuit::GPR tmp = rec.scratch();
+      as.LI    (tmp, instruction.operand_width);
+      as.SUB   (tmp, tmp, result);    // tmp = size - result
```

Comment:
We should be able to do

addi result, -operand size (immediate)
seqz

to save an instruction here